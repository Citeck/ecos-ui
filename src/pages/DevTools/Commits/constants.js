export const ALL_REPOS = 'all';

export const BITBUCKET = 'https://bitbucket.org/';
export const JIRA = 'https://citeck.atlassian.net/browse/';
