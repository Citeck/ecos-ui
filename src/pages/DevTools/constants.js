export const TABS = {
  BUILD: 'build',
  DEV_MODULES: 'dev-modules',
  COMMITS: 'commits',
  SETTINGS: 'settings',
};

export const ECOS_UI_ID = 'ecos-ui';
export const ECOS_UI_LABEL = 'ECOS UI';

export const SETTING_ENABLE_SAGA_LOGGER = 'enableSagaLogger';
export const SETTING_FORCE_ENABLE_NEW_FORMS = 'forceEnableNewForms';
export const SETTING_ENABLE_LOGGING_FOR_NEW_FORMS = 'enableLoggerForNewForms';
export const SETTING_ENABLE_RECORDS_API_DEBUG = 'enableRecordsApiDebug';
export const SETTING_ENABLE_VIEW_NEW_JOURNAL = 'enableViewNewJournal'; // View new journal (2.0)

export default { ECOS_UI_ID, ECOS_UI_LABEL };
