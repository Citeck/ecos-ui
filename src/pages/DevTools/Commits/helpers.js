import { JIRA } from './constants';

const repoRegex = /((git|ssh|http(s)?)|(git@[\w.]+))(:(\/\/)?)([\w.@:/\-~]+)(\.git)(\/)?/;
const taskRegex = /[A-Z]+-[0-9]+/g;

export function getRepoProject(url) {
  const parts = url.match(repoRegex);
  if (!parts) {
    return null;
  }

  return parts[7];
}

export function parseTasksLinks(str) {
  return str.replace(taskRegex, taskId => {
    return `<a href='${JIRA}${taskId}' target='_blank' class='commits-grid__link'>${taskId}</a>`;
  });
}
