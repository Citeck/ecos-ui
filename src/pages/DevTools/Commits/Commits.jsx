import React, { useContext, useEffect } from 'react';
import { Row, Col } from 'reactstrap';

import Select from '../../../components/common/form/Select/Select';
import { t } from '../../../helpers/util';

import { ALL_REPOS } from './constants';
import { CommitsContext } from './CommitsContext';
import CommitsGrid from './CommitsGrid';
import Loader from '../Loader/Loader';
import ErrorText from '../ErrorText/ErrorText';

const CommitsTab = () => {
  const context = useContext(CommitsContext);
  const { state, getCommitsInfo, selectRepo } = context;
  const { isReady, error, repos, repo } = state;

  useEffect(() => {
    getCommitsInfo();
  }, []);

  if (!isReady) {
    return <Loader />;
  } else if (error) {
    return <ErrorText>{error}</ErrorText>;
  }

  const allReposLabel = t('dev-tools.commits.all-repos');
  const options = [
    {
      repo: ALL_REPOS,
      label: allReposLabel
    },
    ...Object.keys(repos).map(id => repos[id])
  ];

  const _selectRepo = option => {
    selectRepo(option.repo);
  };

  return (
    <>
      <Row>
        <Col sm={6} md={4} lg={3}>
          <Select
            options={options}
            getOptionLabel={option => option.label}
            getOptionValue={option => option.repo}
            value={options.find(item => item.repo === repo)}
            onChange={_selectRepo}
            isSearchable={false}
            className={'select_narrow'}
          />
        </Col>
      </Row>
      <CommitsGrid />
    </>
  );
};

export default CommitsTab;
