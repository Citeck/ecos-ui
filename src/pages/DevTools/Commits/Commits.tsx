import React, { useContext, useEffect } from 'react';
// @ts-ignore
import { Row, Col } from 'reactstrap';

import Select from '../../../components/common/form/Select/Select';
import { t } from '../../../helpers/util';
import ErrorText from '../ErrorText/ErrorText';
import Loader from '../Loader/Loader';

import { CommitsContext } from './CommitsContext';
import CommitsGrid from './CommitsGrid';
import { ALL_REPOS } from './constants';

export const Labels = {
  title: 'dev-tools.commits.all-repos'
};

const CommitsTab = (): React.JSX.Element => {
  const context = useContext(CommitsContext);
  const { state, getCommitsInfo, selectRepo } = context;
  const { isReady, error, repos, repo } = state;

  useEffect(() => {
    getCommitsInfo();
  }, []);

  if (error) {
    return <ErrorText>{error}</ErrorText>;
  }

  if (!isReady) {
    return <Loader />;
  }

  const options = [
    {
      repo: ALL_REPOS,
      label: t(Labels.title)
    },
    ...Object.keys(repos).map(id => repos[id])
  ];

  const _selectRepo = (option: { repo: string; label: string }) => {
    selectRepo(option.repo);
  };

  return (
    <>
      <Row>
        <Col sm={6} md={4} lg={3}>
          <Select
            options={options}
            getOptionLabel={(option: { label: string }) => option.label}
            getOptionValue={(option: { repo: string }) => option.repo}
            value={options.find(item => item.repo === repo)}
            onChange={_selectRepo}
            isSearchable={false}
            className={'select_narrow'}
          />
        </Col>
      </Row>
      <CommitsGrid />
    </>
  );
};

export default CommitsTab;
