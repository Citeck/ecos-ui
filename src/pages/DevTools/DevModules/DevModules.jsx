import React, { useContext, useEffect } from 'react';

import PanelTitle from '../../../components/common/PanelTitle/PanelTitle';
import { t } from '../../../helpers/util';

import { DevModulesContext } from './DevModulesContext';
import Loader from '../Loader';
import ErrorText from '../ErrorText';
import DevModulesGrid from './DevModulesGrid';

const DevModulesTab = () => {
  const context = useContext(DevModulesContext);
  const { state, getDevModulesData } = context;
  const { isReady, error } = state;

  useEffect(() => {
    getDevModulesData();
  }, []);

  let devModulesContent = null;
  if (error) {
    devModulesContent = <ErrorText>{error}</ErrorText>;
  } else if (!isReady) {
    devModulesContent = <Loader />;
  } else {
    devModulesContent = <DevModulesGrid />;
  }

  return (
    <>
      <PanelTitle>{t('dev-tools.dev-modules.panel-title')}</PanelTitle>
      {devModulesContent}
    </>
  );
};

export default DevModulesTab;
