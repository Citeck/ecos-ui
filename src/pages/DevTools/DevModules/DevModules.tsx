import React, { useContext, useEffect } from 'react';

import PanelTitle from '../../../components/common/PanelTitle/PanelTitle';
import { t } from '../../../helpers/util';
import ErrorText from '../ErrorText';
import Loader from '../Loader';

import { DevModulesContext } from './DevModulesContext';
import DevModulesGrid from './DevModulesGrid';

export const Labels = {
  title: 'dev-tools.dev-modules.panel-title',
};

const DevModulesTab = (): React.JSX.Element => {
  const context = useContext(DevModulesContext);
  const { state, getDevModulesData } = context;
  const { isReady, error } = state;

  useEffect(() => {
    getDevModulesData();
  }, []);

  let devModulesContent = null;

  if (error) {
    devModulesContent = <ErrorText>{error}</ErrorText>;
  } else if (!isReady) {
    devModulesContent = <Loader />;
  } else {
    devModulesContent = <DevModulesGrid />;
  }

  return (
    <>
      <PanelTitle>{t(Labels.title)}</PanelTitle>
      {devModulesContent}
    </>
  );
};

export default DevModulesTab;
