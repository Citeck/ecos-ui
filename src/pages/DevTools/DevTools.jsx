import React, { useCallback } from 'react';
import { Container, Row, Col } from 'reactstrap';
import queryString from 'query-string';

import PageService from '../../services/PageService';
import Caption from '../../components/common/form/Caption';
import { t } from '../../helpers/util';
import { URL } from '../../constants';

import { TABS } from './constants';
import { DevToolsContextProvider } from './DevToolsContext';
import Tabs from './Tabs';
import TabContent from './TabContent';

import './DevTools.scss';

export default () => {
  const query = queryString.parse(window.location.search);
  let activeTab = query.activeTab;
  if (!Object.values(TABS).includes(activeTab)) {
    activeTab = TABS.BUILD;
  }

  const setActiveTab = useCallback(
    tabId => {
      const query = queryString.parse(window.location.search);
      query.activeTab = tabId;
      const stringQuery = queryString.stringify(query);
      PageService.changeUrlLink(`${URL.DEV_TOOLS}?${stringQuery}`, { updateUrl: true });
    },
    [window.location.search]
  );

  return (
    <div className="dev-tools-page">
      <DevToolsContextProvider activeTab={activeTab} setActiveTab={setActiveTab}>
        <Container>
          <Row>
            <Col>
              <Caption normal className="dev-tools-page__title">
                {t('page-tabs.dev-tools')}
              </Caption>
            </Col>
          </Row>
          <Row>
            <Col>
              <Tabs />
            </Col>
          </Row>
          <Row>
            <Col>
              <TabContent />
            </Col>
          </Row>
        </Container>
      </DevToolsContextProvider>
    </div>
  );
};
