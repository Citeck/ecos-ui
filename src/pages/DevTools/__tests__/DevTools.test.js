import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DevToolsConverter from '../../../dto/devTools';

import { TABS } from '../constants';
import { JIRA, BITBUCKET } from '../Commits/constants';
import { getRepoProject, parseTasksLinks, getHostName } from '../Commits/helpers';
import { input1, output1, input2, output2, input3_4, output3, output4 } from '../__mocks__/DevTools.mock';
import * as DevToolsContext from '../DevToolsContext';
import ErrorText from '../ErrorText';
import Loader from '../Loader';
import Tabs from '../Tabs';
import TabContent from '../TabContent';
import Build from '../Build';
import DevModules from '../DevModules';
import Commits from '../Commits';
import Settings from '../Settings';

configure({ adapter: new Adapter() });

describe('DevTools tests', () => {
  describe('DevToolsConverter', () => {
    it('fetchAlfrescoModulesList', () => {
      expect(DevToolsConverter.fetchAlfrescoModulesList(input1)).toEqual(output1);
    });
    it('fetchSystemModulesList', () => {
      expect(DevToolsConverter.fetchSystemModulesList(input2)).toEqual(output2);
    });
    it('fetchRepos', () => {
      expect(DevToolsConverter.fetchRepos(input3_4)).toEqual(output3);
    });
    it('normalizeCommits', () => {
      expect(DevToolsConverter.normalizeCommits(input3_4)).toEqual(output4);
    });
  });

  describe('Commits helpers', () => {
    it('getRepoProject', () => {
      expect(getRepoProject('git@bitbucket.org:citeck/ecos-ui.git')).toEqual('citeck/ecos-ui');
      expect(getRepoProject('git@bitbucket.org:citeck/ecos-uiserv.git')).toEqual('citeck/ecos-uiserv');
      expect(getRepoProject('git@gitlab.citeck.ru:ecos-ui.git')).toEqual('ecos-ui');
    });
    it('getHostName', () => {
      expect(getHostName(BITBUCKET)).toEqual('bitbucket.org');
      expect(getHostName('git@bitbucket.org:citeck-projects/ecos-ui.git')).toEqual('bitbucket.org');
      expect(getHostName('git@gitlab.citeck.ru:citeck-uiserv/ecos-ui.git')).toEqual('gitlab.citeck.ru');
    });
    it('parseTasksLinks', () => {
      const input1 = 'ECOSCOM-3940 - category-document-type does not have "Delete" and "View in browser" actions';
      const output1 = `<a href='${JIRA}ECOSCOM-3940' target='_blank' class='commits-grid__link'>ECOSCOM-3940</a> - category-document-type does not have "Delete" and "View in browser" actions`;
      expect(parseTasksLinks(input1)).toEqual(output1);

      const input2 = 'Merged into bugfix/ECOSUI-569-2 (pull-request #797)';
      const output2 = `Merged into bugfix/<a href='${JIRA}ECOSUI-569' target='_blank' class='commits-grid__link'>ECOSUI-569</a>-2 (pull-request #797)`;
      expect(parseTasksLinks(input2)).toEqual(output2);
    });
  });

  describe('<Loader />', () => {
    it('should render Loader component', () => {
      const component = shallow(<Loader />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('<ErrorText />', () => {
    it('should render ErrorText component', () => {
      const component = shallow(<ErrorText />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Tabs />', () => {
    let spy = null;
    const renderComponent = (contextValue = {}) => {
      spy = jest.spyOn(DevToolsContext, 'useContext').mockImplementation(() => ({ ...contextValue }));
      const wrapper = shallow(<Tabs />);
      const container = wrapper.find('.dev-tools-page__tabs').at(0);
      const containerProps = container.props();
      const children = containerProps.children;
      return { wrapper, children };
    };

    afterEach(() => {
      spy && spy.mockClear();
    });

    it('should render Tabs component', () => {
      const { wrapper } = renderComponent();
      expect(wrapper).toMatchSnapshot();
    });

    it('no active tab', () => {
      const { children } = renderComponent();
      const tabsItems = children.props.items;
      expect(tabsItems.find(item => item.isActive)).toBeUndefined();
    });

    for (const tabIndex in TABS) {
      it(`check active "${tabIndex}" tab`, () => {
        const { children } = renderComponent({ activeTab: TABS[tabIndex] });
        const tabsItems = children.props.items;
        const activeTab = tabsItems.find(item => item.isActive);
        expect(activeTab.id).toBe(TABS[tabIndex]);
      });
    }

    it(`switch between tabs`, () => {
      const setActiveTab = jest.fn();
      const { children } = renderComponent({ setActiveTab });
      const tabsItems = children.props.items;
      const firstTab = tabsItems[0];
      const lastTab = tabsItems[tabsItems.length - 1];

      expect(setActiveTab).toHaveBeenCalledTimes(0);

      lastTab.onClick();
      expect(setActiveTab).toHaveBeenCalled();
      expect(setActiveTab).toHaveBeenCalledWith(lastTab.id);

      firstTab.onClick();
      expect(setActiveTab).toHaveBeenCalledTimes(2);
      expect(setActiveTab).toHaveBeenNthCalledWith(2, firstTab.id);
    });
  });

  describe('<TabContent />', () => {
    let spy = null;
    const renderComponent = (contextValue = {}) => {
      spy = jest.spyOn(DevToolsContext, 'useContext').mockImplementation(() => ({ ...contextValue }));
      return shallow(<TabContent />);
    };

    afterEach(() => {
      spy && spy.mockClear();
    });

    it('should render TabContent component', () => {
      const wrapper = renderComponent();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render Build component by default', () => {
      const wrapper = renderComponent();
      expect(wrapper.find(Build)).toHaveLength(1);
    });

    it('should render Build component', () => {
      const wrapper = renderComponent({ activeTab: TABS.BUILD });
      expect(wrapper.find(Build)).toHaveLength(1);
    });

    it('should render DevModules component', () => {
      const wrapper = renderComponent({ activeTab: TABS.DEV_MODULES });
      expect(wrapper.find(DevModules)).toHaveLength(1);
    });

    it('should render Commits component', () => {
      const wrapper = renderComponent({ activeTab: TABS.COMMITS });
      expect(wrapper.find(Commits)).toHaveLength(1);
    });

    it('should render Settings component', () => {
      const wrapper = renderComponent({ activeTab: TABS.SETTINGS });
      expect(wrapper.find(Settings)).toHaveLength(1);
    });
  });
});
