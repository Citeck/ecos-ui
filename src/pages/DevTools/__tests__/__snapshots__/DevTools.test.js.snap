// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevTools tests <ErrorText /> should render ErrorText component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "dev-tools-page__error",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "dev-tools-page__error",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`DevTools tests <Loader /> should render Loader component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "blur": true,
      "className": "",
      "color": "light-blue",
      "darkened": false,
      "height": 45,
      "rounded": true,
      "style": Object {
        "margin": "2em 0",
        "position": "relative",
      },
      "type": "circle",
      "width": 45,
      "zIndex": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "blur": true,
        "className": "",
        "color": "light-blue",
        "darkened": false,
        "height": 45,
        "rounded": true,
        "style": Object {
          "margin": "2em 0",
          "position": "relative",
        },
        "type": "circle",
        "width": 45,
        "zIndex": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`DevTools tests <TabContent /> should render TabContent component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BuildTab />,
      "className": "dev-tools-page__tab-content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BuildTab />,
        "className": "dev-tools-page__tab-content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`DevTools tests <Tabs /> should render Tabs component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tabs
        activeTabKey=""
        className=""
        hasHint={false}
        hasHover={false}
        items={
          Array [
            Object {
              "id": "build",
              "isActive": false,
              "label": "dev-tools.tab.build",
              "onClick": [Function],
            },
            Object {
              "id": "dev-modules",
              "isActive": false,
              "label": "dev-tools.tab.dev-modules",
              "onClick": [Function],
            },
            Object {
              "id": "commits",
              "isActive": false,
              "label": "dev-tools.tab.commits",
              "onClick": [Function],
            },
            Object {
              "id": "settings",
              "isActive": false,
              "label": "dev-tools.tab.settings",
              "onClick": [Function],
            },
          ]
        }
        keyField="id"
        narrow={true}
        onClick={null}
        valueField="label"
        valuePrefix=""
      />,
      "className": "dev-tools-page__tabs",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeTabKey": "",
        "className": "",
        "hasHint": false,
        "hasHover": false,
        "items": Array [
          Object {
            "id": "build",
            "isActive": false,
            "label": "dev-tools.tab.build",
            "onClick": [Function],
          },
          Object {
            "id": "dev-modules",
            "isActive": false,
            "label": "dev-tools.tab.dev-modules",
            "onClick": [Function],
          },
          Object {
            "id": "commits",
            "isActive": false,
            "label": "dev-tools.tab.commits",
            "onClick": [Function],
          },
          Object {
            "id": "settings",
            "isActive": false,
            "label": "dev-tools.tab.settings",
            "onClick": [Function],
          },
        ],
        "keyField": "id",
        "narrow": true,
        "onClick": null,
        "valueField": "label",
        "valuePrefix": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tabs
          activeTabKey=""
          className=""
          hasHint={false}
          hasHover={false}
          items={
            Array [
              Object {
                "id": "build",
                "isActive": false,
                "label": "dev-tools.tab.build",
                "onClick": [Function],
              },
              Object {
                "id": "dev-modules",
                "isActive": false,
                "label": "dev-tools.tab.dev-modules",
                "onClick": [Function],
              },
              Object {
                "id": "commits",
                "isActive": false,
                "label": "dev-tools.tab.commits",
                "onClick": [Function],
              },
              Object {
                "id": "settings",
                "isActive": false,
                "label": "dev-tools.tab.settings",
                "onClick": [Function],
              },
            ]
          }
          keyField="id"
          narrow={true}
          onClick={null}
          valueField="label"
          valuePrefix=""
        />,
        "className": "dev-tools-page__tabs",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeTabKey": "",
          "className": "",
          "hasHint": false,
          "hasHover": false,
          "items": Array [
            Object {
              "id": "build",
              "isActive": false,
              "label": "dev-tools.tab.build",
              "onClick": [Function],
            },
            Object {
              "id": "dev-modules",
              "isActive": false,
              "label": "dev-tools.tab.dev-modules",
              "onClick": [Function],
            },
            Object {
              "id": "commits",
              "isActive": false,
              "label": "dev-tools.tab.commits",
              "onClick": [Function],
            },
            Object {
              "id": "settings",
              "isActive": false,
              "label": "dev-tools.tab.settings",
              "onClick": [Function],
            },
          ],
          "keyField": "id",
          "narrow": true,
          "onClick": null,
          "valueField": "label",
          "valuePrefix": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
